packages/api/src/ontology/ActionDefinition.ts

```diff
@@ -14,22 +14,32 @@
  * limitations under the License.
  */
 
-import type { ObjectTypeDefinition } from "../index.js";
 import type { OsdkMetadata } from "../OsdkMetadata.js";
+import type { ObjectTypeDefinition } from "./ObjectTypeDefinition.js";
 
 export interface ActionDefinition<
   A extends string,
   K extends string,
-  T = never, // used in client 2
+  T_signatures = never,
+> {
+  type: "action";
+  apiName: A;
+  osdkMetadata: OsdkMetadata;
+  __DefinitionMetadata?: RawActionDefinition<A, K> & {
+    signatures: T_signatures;
+  };
+}
+
+export interface RawActionDefinition<
```

Comment:
`RawActionDefinition` doesn't feel like the right term here, I want to say `Full` but also understand the trepidations there